### PR TITLE
[Fixes #11] JS Exception on validation in MVC

### LINKED ISSUE #11: JS Exception on validation in MVC
When using with MVC I get this error: "0x800a138f - JavaScript runtime error: Unable to get property 'attr' of undefined or null reference". The problem is in "onSuccess": I don't have "unobtrusiveContainer" defined and it throws this exception.
Since you check whether "unobtrusiveContainer" exists or not via "if (container)" I think it makes sense to move "replaceAttrValue = container.attr("data-valmsg-replace")" under "if (container)".

    function onSuccess(error) {
            var container = error.data("unobtrusiveContainer"),
            replaceAttrValue = container.attr("data-valmsg-replace"),
            replace = replaceAttrValue ? $.parseJSON(replaceAttrValue) : null;

        if (container) {
            container.addClass("field-validation-valid").removeClass("field-validation-error");
            error.removeData("unobtrusiveContainer");

            if (replace) {
                container.empty();
            }
        }
    }

### PR DESCRIPTION
**Pull request by kichalla**
@Eilon @DamianEdwards @pranavkm 

#11

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -10,7 +10,8 @@
         "**/.*",
         "*.json",
         "*.md",
-        "*.txt"
+        "*.txt",
+	"gulpfile.js"
```
Comment (Eilon) [inline bower.json]: Tabs :crying_cat_face:

File: package.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+  "name": "ASP.NET",
+  "version": "0.0.0",
```
Comment (Eilon) [inline package.json]: Should this be the actual version? 3.2.4 or something? @DamianEdwards ?

**Comment by Eilon** [comment]
@DamianEdwards can you look at the gulp/bower stuff?

File: package.json (hunk 2 above)
Comment (DamianEdwards) [inline package.json]: Only if we're actually creating an NPM package, otherwise this is fine.

Comment (Eilon) [inline package.json]: Oh right. Ignore then.

**Comment by DamianEdwards** [comment]
gulp stuff looks good.

We'll need to update `bower.json` with a new version when we actually tag and release this.

**Comment by kichalla** [comment]
good point about `bower.json`...addressed feedback. @DamianEdwards @Eilon  ...I will create a tag once I push this change...also will create a task on templates to use this...

File: jquery.validate.unobtrusive.js [hunk 3]
```diff
@@ -70,11 +70,12 @@
     }
 
     function onSuccess(error) {  // 'this' is the form element
-        var container = error.data("unobtrusiveContainer"),
-            replaceAttrValue = container.attr("data-valmsg-replace"),
-            replace = replaceAttrValue ? $.parseJSON(replaceAttrValue) : null;
+        var container = error.data("unobtrusiveContainer");
 
         if (container) {
+            var replaceAttrValue = container.attr("data-valmsg-replace");
```
Comment (pranavkm) [inline jquery.validate.unobtrusive.js]: We use the format
```JS
var replaceAttrValue = container.attr("data-valmsg-replace"),
    replace = replaceAttrValue ? $.parseJSON(replaceAttrValue) : null;
```
throughout this file. Let's stick to that convention.

**Comment by pranavkm** [comment]
:shipit:

**Comment by kichalla** [comment]
13386cd1b5947d8a5d23a12b531ce3960be1eba7

**Pull request closed without merging**